llm/kv_cache.go

```diff
@@ -0,0 +1,122 @@
+package llm
+
+import (
+	"fmt"
+	"log/slog"
+	"slices"
+
+	"github.com/ollama/ollama/discover"
+	"github.com/ollama/ollama/envconfig"
+)
+
+// Interface for GGML functionality needed by validation
+type GGMLModel interface {
+	KV() KV
+}
+
+// ValidKVCacheTypes contains all supported KV cache types
+var ValidKVCacheTypes = []string{"f32", "f16", "q8_0", "q4_0"}
+
+// modelSupportsFlashAttention checks if the model supports flash attention
+func modelSupportsFlashAttention(ggml GGMLModel) bool {
```

Comment:
We might want to move this into `server.go` since it doesn't have much to do with the kv cache